builder/visualize/padded_stencil_2.html

```diff
@@ -264,26 +264,19 @@
+          {bounds:{min:max((buffer_min(out, 0) + -1), buffer_min(__in, 0)), max:min((buffer_max(out, 0) + 1), buffer_max(__in, 0))}, stride:2, fold_factor:9223372036854775807},
+          {bounds:{min:max((buffer_min(out, 1) + -1), buffer_min(__in, 1)), max:min((buffer_max(out, 1) + 1), buffer_max(__in, 1))}, stride:(((min((buffer_max(out, 0) + 1), buffer_max(__in, 0)) - max((buffer_min(out, 0) + -1), buffer_min(__in, 0))) * 2) + 2), fold_factor:9223372036854775807}
         ]);
         for(let y = (buffer_min(out, 1) + -2); y <= buffer_max(out, 1); y += 1) {
-          { let __intm = crop_buffer(intm, [
-              {min:min(min((buffer_max(out, 0) + 1), max(max((buffer_min(out, 0) + -1), buffer_min(__in, 0)), buffer_max(__in, 0))), max((buffer_min(out, 0) + -1), min(min((buffer_max(out, 0) + 1), buffer_max(__in, 0)), buffer_min(__in, 0)))), max:max(min((buffer_max(out, 0) + 1), max(max((buffer_min(out, 0) + -1), buffer_min(__in, 0)), buffer_max(__in, 0))), max((buffer_min(out, 0) + -1), min(min((buffer_max(out, 0) + 1), buffer_max(__in, 0)), buffer_min(__in, 0))))},
-              {min:min(min((y + 1), max(min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1)), max((y + -1), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1))))), max(max((y + -1), min(min((y + 1), min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1))), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1)))), max((y + 1), min(min((y + 1), min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1))), max((y + -1), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1))))))), max:max(min((y + 1), max(min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1)), max((y + -1), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1))))), min(max((y + -1), min(min((y + 1), min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1))), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1)))), max((y + 1), min(min((y + 1), min(max((buffer_max(out, 1) + 1), max((buffer_min(out, 1) + -1), buffer_min(__in, 1))), buffer_max(__in, 1))), max((y + -1), max(min((buffer_min(out, 1) + -1), min((buffer_max(out, 1) + 1), buffer_max(__in, 1))), buffer_min(__in, 1)))))))}
-          ]);
+          { let __intm = crop_dim(intm, 1, {min:(max(y, (max(y, buffer_min(out, 1)) + -2)) + 1), max:(y + 1)});
```

Comment:
Awesome... I've looked at a lot at this expression trying to simplify it. This is a lot more simpliflied than I ever thought it could get!